source/extensions/filters/network/http_connection_manager/config.h

```diff
@@ -314,6 +316,7 @@ class HttpConnectionManagerConfig : Logger::Loggable<Logger::Id::config>,
   std::chrono::milliseconds request_headers_timeout_;
   Router::RouteConfigProviderSharedPtr route_config_provider_;
   Config::ConfigProviderPtr scoped_routes_config_provider_;
+  Router::ScopeKeyBuilderPtr scope_key_builder_;
```

Comment:
Can you explain (in a comment) what's the relation between the `scoped_routes_config_provider_` and `scope_key_builder_`? Specifically I wonder if only one can be non-null and the other will.